### PR TITLE
Updated yaml files

### PR DESCRIPTION
**Pull request by pradeepachar-mavenir**
- Updated RACM and CallHandling specifications.
- Notification Channel specification are added.

#### What type of PR is this?

Add one of the following kinds:
* bug
* correction
* enhancement/feature
* cleanup
* documentation
* subproject management
* tests

#### Which issue(s) this PR fixes:

Fixes #

#### Changelog input

```
 release-note

```

#### Additional documentation 

This section can be blank.

```
docs

```

### CODE REVIEW

**Comment by pradeepachar-mavenir** [comment]
@sushanthmavenir @TEF-RicardoSerr , please review the pull request.

**Comment by TEF-RicardoSerr** [comment]
Hello, 

How would it fit the integration of other notification methods in the specifications? In the previous versions of the API, PNS was integrated in the RACM specification but it is no longer in any of these new specs. Would it be possible to include PNS in the new specification?

**Comment by pradeepachar-mavenir** [comment]
> Hello,
> 
> How would it fit the integration of other notification methods in the specifications? In the previous versions of the API, PNS was integrated in the RACM specification but it is no longer in any of these new specs. Would it be possible to include PNS in the new specification?

In the new notification API, there is channelType defined, this can be extended to support new notification channels as requried.

**Comment by TEF-RicardoSerr** [comment]
Do you find it feaseable to include PNS in this specification @pradeepachar-mavenir?

**Comment by deepakjaiswal1** [comment]
@pradeepachar-mavenir We should eliminate the use of userId within the CallHandling APIs, as users can be identified through the access/identification token.

Additionally, we should consider removing the responseCode from SessionInvitationNotification, SessionStatusNotification, and ReceiverStatusNotification. This information is redundant and could cause confusion for developers using the endpoints.

**Comment by deepakjaiswal1** [comment]
@pradeepachar-mavenir @TEF-RicardoSerr 

We firmly believe that the Channel service provider could function as an independent service rather than being integrated into the WebRTC GW. In our perspective, the RACM & channel APIs should follow this call-flow:

A) The client app initiates the creation of a channel (WebSocket).

POST /channels 
{
    "notificationChannel": {
        "channelType": "WebSockets",
        "channelData": {
            "maxNotifications": "1"
        },
        "clientCorrelator": "xyz123"
    }
}

HTTP/1.1 201 Created
{
    "notificationChannel": {
        "callbackURL": "http://example.com/callBackUrl/cbu112",
        "channelData": {
            "channelURL": "ws://example.com/exampleAPI/notificationchannel/v1/tel%3A%2B19585550100/channels/ch378/ws",
            "maxNotifications": "1"
        },
        "channelType": "WebSockets",
        "clientCorrelator": "xyz123",
        "resourceURL": "http://example.com/exampleAPI/notificationchannel/v1/tel%3A%2B19585550100/channels/ch378"
    }
}

B) Subsequently, the client app establishes an active session, providing the callback for the previously created notification channel.

POST /session
{
    "deviceId": "string",
    "notificationChannel": {
        "callbackURL": "http://example.com/callBackUrl/cbu112"
    }
}

HTTP/1.1 201 Created
{
    "clientId": "string",
    "resourceURL": "string"
}

The WebRTC GW posts client notifications using the callbackURL. By adopting this framework, the WebRTC GW is relieved from managing the intricacies of different notification channels. However, we need to consider cases where clients utilize multiple channels, such as a PNS channel for initial notification and a WebSocket channel for subsequent notifications.

We have two options to address this:
(A) Implement a combined channel for PNS + WebSocket, where the channel itself manages the logic to send notifications through the appropriate channel.
(B) Utilize multiple channels alongside a subscription model, where the client subscribes to specific events over each channel. We recommend option A.

**Comment by pradeepachar-mavenir** [comment]
> @pradeepachar-mavenir We should eliminate the use of userId within the CallHandling APIs, as users can be identified through the access/identification token.
> 
> Additionally, we should consider removing the responseCode from SessionInvitationNotification, SessionStatusNotification, and ReceiverStatusNotification. This information is redundant and could cause confusion for developers using the endpoints.

**Comment by pradeepachar-mavenir** [comment]
> #### Closed with unmerged commits

**Comment by deepakjaiswal1** [comment]
@pradeepachar-mavenir Appreciate the updated YAML files. The NotificationChannel schema lacks the necessary callbackURL for posting notifications on the channel. Could we include the callbackURL in the notification channel schema?

**Comment by pradeepachar-mavenir** [comment]
> @pradeepachar-mavenir Appreciate the updated YAML files. The NotificationChannel schema lacks the necessary callbackURL for posting notifications on the channel. Could we include the callbackURL in the notification channel schema?

@deepakjaiswal1 In the RACM API, a new element notificationUri is added. The clients are expected to establish the notification channel with the notification server and include the callback uri provided by notification server in the notificationUri attribute.

File: code/API_definitions/BYON-RACM-Service.yaml [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+              $ref: '#/components/schemas/RacmRequest'
+      responses:
+        '200':
+          description: Ok
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RacmResponse'
+        '400':
+          $ref: '#/components/responses/Generic400'
```
Comment (TEF-RicardoSerr) [inline code/API_definitions/BYON-RACM-Service.yaml:41]: I suggest to include the Error codes defined by Deepak

```suggestion
        '400':
          description: Bad Request
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ErrorInfo'
              examples:
                400emergencyAdressMissing:
                  summary: Emergency address missing
                  value:
                    status: 400
                    code: EMERGENCY_ADDRESS_MISSING
                    message: Emergency address missing
                400genericResponse:
                  summary: Generic Response
                  value:
                    status: 400
                    code: INVALID_ARGUMENT
                    message: Schema validation failed at  ...
        '403':
          description: Forbidden
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ErrorInfo'
              examples:
                403maxRegistrationUser:
                  summary: Max registration per user
                  value:
                    status: 403
                    code: MAX_REGISTRATION_PER_USER
                    message: 'The maximum number of service registrations per user has been reached'
                403maxRegistrationMSISDN:
                  summary: Max registration per MSISDN
                  value:
                    status: 403
                    code: MAX_REGISTRATION_PER_MSISDN
                    message: 'The maximum number of service registrations per MSISDN has been reached'
                403maxDevicesUser:
                  summary: Max devices per user
                  value:
                    status: 403
                    code: MAX_DEVICES_PER_USER
                    message: 'The maximum number of device instances per user has been reached'
                403genericResponse:
                  summary: Generic Response
                  value:
                    status: 403
                    code: PERMISSION_DENIED
                    message: 'Operation not allowed: ...' 
```

**Review by TEF-RicardoSerr — CHANGES_REQUESTED** [review]
Hello, just a few comments regarding the versioning and include o deepack error codes

File: code/API_definitions/BYON-Notification-Channel.yaml [hunk 2]
```diff
@@ -0,0 +1,246 @@
+openapi: 3.0.0
+info:
+  title: Bring Your Own Number (BYON) Notification Channel APIs
+  description: This API provide REST API for a client, browser or native application, to establish notification channel to receive asynchronous notifications from MNO's IMS Network.
+  version: 1.0.0
```
[the remark this answers is not in the document]
Comment (pradeepachar-mavenir) [reply code/API_definitions/BYON-Notification-Channel.yaml:5]: version changed to 0.1.0

**Review by TEF-RicardoSerr — APPROVED** [review]
we can come up with the first version

**Pull request merged**